scripts/pcm-dpc/covid-19/preprocess.py

```diff
@@ -0,0 +1,193 @@
+                            'master/scripts/istat/geos/cleaned/ISTAT_region.csv')
+        region_code = pd.read_csv(region_code_path)[['Region Code', 'NUTS2']]
+        code_dict = region_code.set_index('Region Code').to_dict()['NUTS2']
+        # region code 21 and 22 is missing from the dict above, add manually here.
+        code_dict[21] = 'nuts/ITH1'
+        code_dict[22] = 'nuts/ITH2'
+        self.data['Location'] = self.data['RegionCode'].map(code_dict)
+        self.data = self.data.drop(columns=['State', 'RegionCode', 'RegionName',
+                                            'Latitude', 'Longitude'])
+
+    def geo_template(self):
+        GEO_TEMPLATE = 'Node: E:pcm-dpc->E0\n' +\
+                    'typeOf: dcs:EurostatNUTS2\n' +\
+                    'dcid: C:pcm-dpc->Location\n\n'
+        with open(self.name + '.tmcf', 'w') as f_out:
+            f_out.write(GEO_TEMPLATE)
+        return 'E:pcm-dpc->E0'
+
+
+class PcmDpcProvinces(PcmDpc):
+    """subclass of provinces data"""
+    def __init__(self):
+        self.csvpath = ('https://raw.githubusercontent.com/pcm-dpc/COVID-19/'
+                        'master/dati-province/dpc-covid19-ita-province.csv')
+        self.name = "dpc-covid19-ita-province"
+        self.stat_vars = ['CumulativeCount_MedicalConditionIncident_'
+                          'COVID_19_PositiveCase']
+
+    def set_location(self):
+        province_code_path = ('https://raw.githubusercontent.com/qlj-lijuan/data'
+                              '/master/scripts/istat/geos/cleaned/ISTAT_province'
+                              '.csv')
+        province_code = pd.read_csv(province_code_path)[['Province Abbreviation',
+                                                         'NUTS3']]
+        province_dict = (province_code.set_index('Province Abbreviation')
+                         .to_dict()['NUTS3'])
+        # drop the data whose location is "being defined/updated",
+        # i.e. ProvinceCode > 111.
+        # location Sud Sardegna (Province Code = 111) is not defined as a
+        # unique area in DataCommons, skip for now.
+        self.data = self.data[self.data['ProvinceCode'] < 111].reset_index()
+        self.data['Location'] = self.data['ProvinceAbbreviation'].map(
+            province_dict)
+        self.data = self.data[['Date', 'Location', 'CumulativeCount_Medical'
+                               'ConditionIncident_COVID_19_PositiveCase']]
+
+    def geo_template(self):
+        GEO_TEMPLATE = ('Node: E:pcm-dpc->E0\n'
+                        'typeOf: dcs:EurostatNUTS3\n'
+                        'dcid: C:pcm-dpc->Location\n\n')
+        with open(self.name + '.tmcf', 'w') as f_out:
+            f_out.write(GEO_TEMPLATE)
+        return 'E:pcm-dpc->E0'
+
+def main():
+    """process the national, regional, provinces data and generate
+    correponding template mcfs"""
+    data_obj = [PcmDpcNational(), PcmDpcRegions(), PcmDpcProvinces()]
+    for data in data_obj:
```

Comment:
```suggestion
    for data in [PcmDpcNational(), PcmDpcRegions(), PcmDpcProvinces()]:
```

The typical rule of thumb is to avoid defining a variable that is used once, especially one used immediately after the definition.  There are exceptions, e.g., avoiding long lines.

Reply:
Got it. Thanks